static/js/hydrate.js

```diff
@@ -145,7 +145,7 @@ function hydrateSpecies(){
             hydrateNextEvolutionWithMoves(i, evo)
         }
         // prepare to be appended a list of location where this pokemon appear
-        spec.locations = [];
+        spec.locations = new Map();
         // add to the html list 
         const core = document.createElement('div')
```

Comment:
the main reason i criticise this map choice is because i don't use map for any other types
I may be in the wrong to want to push out the array but like since you index it with map ID at the end??